provision-osx/tasks/install_xcode.yml

```diff
@@ -1,68 +1,91 @@
 ---
 
 -
-  name: Check if ruby_domain_name gem is already installed
-  shell: "gem list -i '^domain_name$'"
-  args:
-    executable: /bin/bash
+  name: Get xcversion binary path
+  shell: source ~/.bash_profile && which xcversion
+  register: xcversion_bin
   changed_when: False
-  failed_when: ruby_domain_name_installed.stderr and 'command not found' in ruby_domain_name_installed.stderr
-  register: ruby_domain_name_installed
+  failed_when: xcversion_bin.rc != 0
 
 -
-  name: Copy ruby_domain_name gem to remote host
-  copy:
-     src: "{{ruby_domain_name_file_path}}"
-     dest: "{{remote_tmp_dir}}/{{ruby_domain_name_gem_name}}"
-     mode: 0755
-  when: ruby_domain_name_installed.rc != 0
-
--
-  name: Install ruby_domain_name gem
-  shell: "gem install {{remote_tmp_dir}}/{{ruby_domain_name_gem_name}}"
+  name: Install command line tools
+  shell: "{{ xcversion_bin.stdout }} install-cli-tools"
   args:
     executable: /bin/bash
   become: yes
-  when: ruby_domain_name_installed.rc != 0
+  register: xcode_cli_tools_install
+  when: xcode_install_user is defined and xcode_install_password is defined
+  failed_when: xcode_cli_tools_install.rc != 0 and xcode_cli_tools_install.stdout.find('already installed') == -1
+  changed_when: xcode_cli_tools_install.stdout.find('already installed') == -1
 
 -
-  name: Install xcode-install gem package
-  shell: "gem install --conservative --no-ri --no-rdoc xcode-install -v {{ xcode_install_version | default('\"\"') }}"
-  args:
-    executable: /bin/bash
+  name: Get list of available xcode versions
+  command: "{{ xcversion_bin.stdout }} list --all"
+  register: xcode_available_status
+  changed_when: False
   become: yes
   environment:
-    http_proxy: "{{ proxy_url | default('') }}"
-    https_proxy: "{{ proxy_url | default('') }}"
+    XCODE_INSTALL_USER: "{{ xcode_install_user }}"
+    XCODE_INSTALL_PASSWORD: "{{ xcode_install_password }}"
 
 -
-  name: Install command line tools
-  shell: source ~/.bash_profile && xcversion install-cli-tools
-  args:
-    executable: /bin/bash
-  register: xcode_cli_tools_install
-  when: xcode_install_user is defined and xcode_install_password is defined
-  failed_when: xcode_cli_tools_install.rc != 0 and xcode_cli_tools_install.stdout.find('already installed') == -1
-  changed_when: xcode_cli_tools_install.stdout.find('already installed') == -1
+  name: Update cached xcode versions for {{ ansible_user }}
+  command: "{{ xcversion_bin.stdout }} update"
+  become: yes
   environment:
     XCODE_INSTALL_USER: "{{ xcode_install_user }}"
     XCODE_INSTALL_PASSWORD: "{{ xcode_install_password }}"
-    FASTLANE_SESSION: "{{ xcode_install_session_token | default(omit) }}"
```

Comment:
Should this be added back in where required as I see we have it in our current docs. Ping @odra

Reply:
yeah let me add it back